frontend/src/components/SignInPopup/SignInPopup.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import PopupWithForm from '../PopupWithForm/PopupWithForm';
+import { useFormWithValidation } from '../../utils/Validation';
+
+function SignInPopup(props) {
+  const { errors, handleChange, isValid, resetForm, values } =
+    useFormWithValidation();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    props.onLogin(values);
+    resetForm();
```

Comment:
`    resetForm();
`

You need to **reset** the form only when you **open** it not to do that immediately after `submit` because when there is an error the popup will still be opened and the user can try to send the data again to the server without having to print the data into the inputs once more


```js
    React.useEffect(() => {
         resetForm();
    }, [isOpen]);

```